WorkloadManagementSystem/DB/JobDB.py

```diff
@@ -796,135 +799,135 @@ def setJobParameters( self, jobID, parameters ):
+  def removeJobOptParameter(self, jobID, name):
     """ Remove the specified optimizer parameter for jobID
     """
-    ret = self._escapeString( jobID )
+    ret = self._escapeString(jobID)
     if not ret['OK']:
       return ret
     jobID = ret['Value']
-    ret = self._escapeString( name )
+    ret = self._escapeString(name)
     if not ret['OK']:
       return ret
     name = ret['Value']
 
-    cmd = 'DELETE FROM OptimizerParameters WHERE JobID=%s AND Name=%s' % ( jobID, name )
-    if not self._update( cmd )['OK']:
-      return S_ERROR( 'JobDB.removeJobOptParameter: operation failed.' )
+    cmd = 'DELETE FROM OptimizerParameters WHERE JobID=%s AND Name=%s' % (jobID, name)
+    if not self._update(cmd)['OK']:
+      return S_ERROR('JobDB.removeJobOptParameter: operation failed.')
     return S_OK()
 
 #############################################################################
-  def setAtticJobParameter( self, jobID, key, value, rescheduleCounter ):
+  def setAtticJobParameter(self, jobID, key, value, rescheduleCounter):
     """ Set attic parameter for job specified by its jobID when job rescheduling
         for later debugging
     """
-    ret = self._escapeString( jobID )
+    ret = self._escapeString(jobID)
     if not ret['OK']:
       return ret
     jobID = ret['Value']
 
-    ret = self._escapeString( key )
+    ret = self._escapeString(key)
     if not ret['OK']:
       return ret
     key = ret['Value']
 
-    ret = self._escapeString( value )
+    ret = self._escapeString(value)
     if not ret['OK']:
       return ret
     value = ret['Value']
 
-    ret = self._escapeString( rescheduleCounter )
+    ret = self._escapeString(rescheduleCounter)
     if not ret['OK']:
       return ret
     rescheduleCounter = ret['Value']
 
     cmd = 'INSERT INTO AtticJobParameters (JobID,RescheduleCycle,Name,Value) VALUES(%s,%s,%s,%s)' % \
-         ( jobID, rescheduleCounter, key, value )
-    result = self._update( cmd )
+        (jobID, rescheduleCounter, key, value)
+    result = self._update(cmd)
     if not result['OK']:
-      result = S_ERROR( 'JobDB.setAtticJobParameter: operation failed.' )
+      result = S_ERROR('JobDB.setAtticJobParameter: operation failed.')
 
     return result
 
 #############################################################################
-  def __setInitialJobParameters( self, classadJob, jobID ):
+  def __setInitialJobParameters(self, classadJob, jobID):
     """ Set initial job parameters as was defined in the Classad
     """
 
     # Extract initital job parameters
     parameters = {}
-    if classadJob.lookupAttribute( "Parameters" ):
-      parameters = classadJob.getDictionaryFromSubJDL( "Parameters" )
-    res = self.setJobParameters( jobID, parameters.items() )
+    if classadJob.lookupAttribute("Parameters"):
+      parameters = classadJob.getDictionaryFromSubJDL("Parameters")
+    res = self.setJobParameters(jobID, list(parameters.items()))
```

Comment:
Was this change done automatically? Cf. https://github.com/DIRACGrid/DIRAC/pull/3763#discussion_r205439203

Reply:
Yes, enclosing `parameters.items()` in a `list` was automatically done with the `python-modernize` tool.

Reply:
Then I don't fully get it: at line 1240 of this same module there is `for param, val in voPolicy.items():` and that line was not touched. It seems that when there's a loop on `.items()` results, python-modernize ignores it. This is confusing and IMHO dangerous.

Reply:
Yes, that seems weird (regarding changes). Let me explore a bit on this and can be discussed later.